File: src/main/java/gg/mineral/server/util/json/JsonUtil.java
package gg.mineral.server.util.json;

import java.io.IOException;
import java.net.MalformedURLException;

import org.json.JSONObject;

import gg.mineral.server.network.http.HttpsClient;
import gg.mineral.server.network.http.MimeType;
import lombok.SneakyThrows;
import lombok.val;

public class JsonUtil {

    @SneakyThrows
    public static JSONObject getJsonObject(String url) {
        val response = doGetCall(url);
        return response == null ? null : new JSONObject(response);
    }

    private static String doGetCall(String url)
            throws IOException, MalformedURLException {
        return new HttpsClient(url, HttpsClient.Mode.GET).setAcceptLanguage("en_US")
                .setAccept(MimeType.JSON).send();
    }
}

File: server/src/main/java/gg/mineral/server/util/json/JsonUtil.kt
package gg.mineral.server.util.json

import gg.mineral.server.network.http.HttpsClient
import gg.mineral.server.network.http.MimeType
import lombok.SneakyThrows
import org.json.JSONObject
import java.io.IOException
import java.net.MalformedURLException

object JsonUtil {
    @SneakyThrows
    fun getJsonObject(url: String): JSONObject {
        val response = doGetCall(url)
        return JSONObject(response)
    }

    @Throws(IOException::class, MalformedURLException::class)
    private fun doGetCall(url: String): String {
        return HttpsClient(url, HttpsClient.Mode.GET).setAcceptLanguage("en_US")
            .setAccept(MimeType.JSON).send()
    }
}
